core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -38,31 +39,49 @@ object Helper {
   def readFileAsync(path: String): IO[List[String]] =
     IO(Source.fromFile(path, "UTF-8")).bracket(source => IO.pure(source.getLines.toList))(source => IO(source.close()))
 
-  def hideSymbols(str: String): String = {
-    str.foldRight("") {
-      case (c, res) if SPECIAL_CHARS contains c =>
-        s"\\$c$res"
-      case (c, res) =>
-        s"$c$res"
+  def preciseMatch(query: String): String = {
+    val queryTokens: Seq[Token] = Tokenizer.parseStringWithSpecialSymbols(query)
+    val preciseMatch: Seq[Token] = queryTokens.map {
+      case SpecialSymbol(value @ "|") => Escaped(value.charAt(0))
```

Comment:
Maybe it would be better to have a character set like `Set('^',  '$',  '+', '*' etc...)` and check character set with pattern matching?

Reply:
I have no idea how it will look. Can you give a little example?
I think that now can make TypeAlias over "SpecailSymbol", which will be a "Spec" or "Symbol"ю

Reply:
![image](https://user-images.githubusercontent.com/5605372/49698563-ed755000-fbde-11e8-8e41-2ef8c2480128.png)

Reply:
Okay, will make.